From 1981b428c14ad6790b097235063f2ce0e9d57631 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Wed, 3 Sep 2025 11:26:55 +0200
Subject: [PATCH 072/133] Revert "test: disable non-native loongarch64 behavior
 and std tests"
X-Developer-Signature: v=1; a=openpgp-sha256; l=1183; i=xtex@astrafall.org;
 h=from:subject; bh=EbFMZd9bDVGoEvPDfuuA28mxuRpkf6LalJiUSeEW68Q=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78lhfZPVPapw7Y5bC00NFJ45+irKPYvDzPRGwhdVh7
 15zft0fHRNZGMS4GCzFFFmKDBu8WXXS+UWXlcvCzGFlAhkiLdLAwMDAwMLAl5uYV2qkY6Rnqm2o
 Z2ioY6RjxMDFKQBTbfmE4X/44kO+D7bzFV2+aF3r0H/pqMzHDaHSzJNYpEs6l0tNWPybkeHM6vW
 /txqnnI+sWyVdNz3kgVt372mvt08mrrzq8rI09iUjAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

This reverts commit f4ed35f800396f12c7cd6aa1f70cf2555ddf7c84.

This should no longer be needed now that stack traces are disabled on loongarch.

closes #24405

(cherry picked from commit b2ef6d01fff852f2f1b24bba4917055f4e699f19)
---
 test/tests.zig | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/test/tests.zig b/test/tests.zig
index 6b14685a6f84..637c7611ca39 100644
--- a/test/tests.zig
+++ b/test/tests.zig
@@ -2296,11 +2296,6 @@ pub fn addModuleTests(b: *std.Build, options: ModuleTestOptions) *Step {
         if (options.skip_single_threaded and test_target.single_threaded == true)
             continue;
 
-        // https://github.com/ziglang/zig/issues/24405
-        if (!builtin.cpu.arch.isLoongArch() and target.cpu.arch.isLoongArch() and
-            (mem.eql(u8, options.name, "behavior") or mem.eql(u8, options.name, "std")))
-            continue;
-
         // TODO get compiler-rt tests passing for self-hosted backends.
         if (((target.cpu.arch != .x86_64 and target.cpu.arch != .aarch64) or target.ofmt == .coff) and
             test_target.use_llvm == false and mem.eql(u8, options.name, "compiler-rt"))
-- 
2.52.0

